### ISSUE TITLE
Hang/Crash On Autocraft Attempt

### ISSUE THREAD

**Issue by dannydjdk** [raoulvdberge/refinedstorage#608]
#### Issue description:
While attempting to autcraft a recipe for an Exporter with a crafting card, the server hangs until crash, single player game hangs until minecraft process is killed. I have not been able to isolate all conditions where this does or does not occur, but at the time of the crashes, it was attempting to craft Canola Seeds from Actually Additions from Canola when there was no Canola in the system. The crafting process is triggered by an Exporter with a crafting card.
The world, the autocrafting recipe and the exporter were all set up prior to version 1.2. Reverting to 1.1.2 version does fix the issue. However, on upgrading to 1.2+ (including 1.2.3), the world will always hang after a few minutes of play (presumably as soon as it attempts to craft the Canola Seeds).

#### What happens:
Game becomes unresponsive while autocrafting task hangs in the background. When running on a server, the crash occurs. Crash report link below.

#### What you expected to happen:
Game play continues even if the autocrafting process does not have the resources it needs or some other problem occurs with the recipe.

#### Steps to reproduce:
1. Set up simple autocrafting recipe.
2. Set up exporter to export results of recipe.
3. Deprive the system of an item needed for the crafting recipe.

#### Version (Make sure you are on the latest version before reporting):

- Minecraft: 1.10.2
- Forge: 12.18.2.2123
- Refined Storage: 1.2.3

Does this issue occur on a server? yes

#### If a (crash)log is relevant for this issue, link it here:

http://pastebin.com/ntmavXzV

**Comment by Kris6699** [comment]
Could be same as me: #607

**Comment by dannydjdk** [comment]
I found a workaround which may or may not make this issue obsolete.

1. I rolled back to 1.1.2 and removed all crafting upgrades from exporters.
2. Upgraded to 1.2.4 and broke and replaced the exporters.
3. Reinserted the crafting cards.

FYI: I attempted to reinsert the crafting cards without braking and replacing the exporters and the game immediately stopped responding to clicks and the server crashed. This was not isolated to Canola seeds but also included a vanilla stone axe recipe.
I also had to break and replace all my Detectors as they no longer would output redstone signals when appropriate, and the GUI buttons did not respond to clicks. I also had to break and replace one of the grids as it could not be turned on. Breaking and replacing the controller did not fix these issues.

In short, all these issues only affected blocks that were placed prior to the upgrade. Breaking and replacing each affected block fixed the issues. However, a roll back to the previous version was necessary to work around the game breaking hang that occurred when attempting to craft any recipe requested via a crafting upgrade.

**Comment by way2muchnoise** [comment]
When the watchdog kills the server the logs are kinda useless. If you reproduce this 100%, you might want to create a profiling thing see https://github.com/raoulvdberge/refinedstorage/issues/523#issuecomment-256419502

**Comment by raoulvdberge** [comment]
Closing due to missing profling report. If you send us a report, I'll reopen
